### PR TITLE
DOC: Avoid using "set of" when talking about an ordered list.

### PR DESCRIPTION
**Pull request by anntzer**
... or when the input isn't/cannot be a set.  I left a few usages, e.g.
in random sampling, where "set" is reasonable as informal description of
an array as the order doesn't matter; however, for e.g. np.gradient the
order of the returned list is clearly important, so "set" is wrong.

Also some other minor doc edits noticed during the grepping: using
`shape` instead of `form` in `cov` is consistent with most other places;
the wording in `Polynomial.trim` now matches other methods on the same
class.

### CODE REVIEW

File: benchmarks/benchmarks/bench_app.py [hunk 1]
```diff
@@ -70,8 +70,8 @@ def maxes_of_dots(self, arrays):
 
         Arrays must agree only on the first dimension.
 
-        For numpy it a join benchmark of dot products and max()
-        on a set of arrays.
+        For numpy it a joint benchmark of dot products and
```
Comment (charris) [inline benchmarks/benchmarks/bench_app.py:73]: Needs "is" before "a".

File: benchmarks/benchmarks/bench_app.py [hunk 2]
```diff
@@ -70,8 +70,8 @@ def maxes_of_dots(self, arrays):
...
+        max(<array>, axis=<int>).
         """
```
Comment (charris) [inline benchmarks/benchmarks/bench_app.py:75]: I don't find this more enlightening. What does the benchmark actually do?

**Comment by charris** [comment]
Note that "set", "class", "collection" have colloquial meanings that don't exactly align with mathematical usage.

**Comment by anntzer** [comment]
> Note that "set", "class", "collection" have colloquial meanings that don't exactly align with mathematical usage.

I agree, that's why I left many usages ("set of bins" for histograms, "set of choices" for random sampling) when it clearly didn't matter.

File: benchmarks/benchmarks/bench_app.py (hunk 2 above)
Comment (anntzer) [reply benchmarks/benchmarks/bench_app.py:75]: reworded

**Comment by charris** [comment]
Thanks @anntzer .

**Pull request merged**